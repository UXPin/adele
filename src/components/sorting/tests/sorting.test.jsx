import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import Sorting from '../sorting';

const mockFunc = () => true;
const testString = 'test-string';

it('renders sorter AZ correctly', () => {
  const wrapper = mount(<Sorting sort={mockFunc} sorting="az" activeSorter={testString} category={testString} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders sorter ZA correctly', () => {
  const wrapper = mount(<Sorting sort={mockFunc} sorting="za" activeSorter={testString} category={testString} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders sorter default correctly', () => {
  const wrapper = mount(<Sorting sort={mockFunc} sorting="def" activeSorter={testString} category={testString} />);
  expect(wrapper).toMatchSnapshot();
});
