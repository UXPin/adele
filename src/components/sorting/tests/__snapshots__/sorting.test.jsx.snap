// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sorter AZ correctly 1`] = `
.c1 {
  background: none;
  border: none;
  outline: none;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  color: #006cff;
  cursor: pointer;
  padding-top: 3px;
  padding-bottom: 0;
  margin-top: 0;
}

.c1:hover,
.c1:focus {
  color: #006cff;
}

.c1 span {
  position: relative;
  top: 1px;
  left: -2px;
}

.c0 {
  width: 20px;
  height: 100%;
  padding-top: 0;
  margin: 0;
}

.c2:focus {
  outline: 0;
}

.c2 > svg {
  width: 7px;
  height: 100%;
  fill: #006cff;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  cursor: pointer;
}

.c2 > svg:hover {
  fill: #006cff;
}

<Sorting
  activeSorter="test-string"
  category="test-string"
  sort={[Function]}
  sorting="az"
  tab={0}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.button
        activeSorter="test-string"
        category="test-string"
        onClick={[Function]}
        sorting="az"
        tabIndex={0}
        title="Current sorting: test-string az"
      >
        <button
          className="c1"
          onClick={[Function]}
          tabIndex={0}
          title="Current sorting: test-string az"
        >
          A/Z
          <Icon
            active={true}
            color="#006cff"
            i="test-file-stub"
            in="blue"
            rotate={180}
            size="xs"
          >
            <Styled(SVGInline)
              color="#006cff"
              data-active={true}
              in="blue"
              rotate={180}
              size="xs"
              svg="test-file-stub"
            >
              <SVGInline
                className="c2"
                classSuffix="-svg"
                cleanup={Array []}
                cleanupExceptions={Array []}
                color="#006cff"
                component="span"
                data-active={true}
                in="blue"
                rotate={180}
                size="xs"
                svg="test-file-stub"
              >
                <span
                  className="SVGInline c2"
                  color="#006cff"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "test-file-stub",
                    }
                  }
                  data-active={true}
                  in="blue"
                  rotate={180}
                  size="xs"
                />
              </SVGInline>
            </Styled(SVGInline)>
          </Icon>
        </button>
      </styled.button>
    </div>
  </styled.div>
</Sorting>
`;

exports[`renders sorter ZA correctly 1`] = `
.c1 {
  background: none;
  border: none;
  outline: none;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  color: #006cff;
  cursor: pointer;
  padding-top: 3px;
  padding-bottom: 0;
  margin-top: 0;
}

.c1:hover,
.c1:focus {
  color: #006cff;
}

.c1 span {
  position: relative;
  left: -2px;
}

.c0 {
  width: 20px;
  height: 100%;
  padding-top: 0;
  margin: 0;
}

.c2:focus {
  outline: 0;
}

.c2 > svg {
  width: 7px;
  height: 100%;
  fill: #006cff;
  cursor: pointer;
}

.c2 > svg:hover {
  fill: #006cff;
}

<Sorting
  activeSorter="test-string"
  category="test-string"
  sort={[Function]}
  sorting="za"
  tab={0}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.button
        activeSorter="test-string"
        category="test-string"
        onClick={[Function]}
        sorting="za"
        tabIndex={0}
        title="Current sorting: test-string za"
      >
        <button
          className="c1"
          onClick={[Function]}
          tabIndex={0}
          title="Current sorting: test-string za"
        >
          Z/A
          <Icon
            active={true}
            color="#006cff"
            i="test-file-stub"
            in="blue"
            rotate={0}
            size="xs"
          >
            <Styled(SVGInline)
              color="#006cff"
              data-active={true}
              in="blue"
              rotate={0}
              size="xs"
              svg="test-file-stub"
            >
              <SVGInline
                className="c2"
                classSuffix="-svg"
                cleanup={Array []}
                cleanupExceptions={Array []}
                color="#006cff"
                component="span"
                data-active={true}
                in="blue"
                rotate={0}
                size="xs"
                svg="test-file-stub"
              >
                <span
                  className="SVGInline c2"
                  color="#006cff"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "test-file-stub",
                    }
                  }
                  data-active={true}
                  in="blue"
                  rotate={0}
                  size="xs"
                />
              </SVGInline>
            </Styled(SVGInline)>
          </Icon>
        </button>
      </styled.button>
    </div>
  </styled.div>
</Sorting>
`;

exports[`renders sorter default correctly 1`] = `
.c1 {
  background: none;
  border: none;
  outline: none;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  color: #777777;
  cursor: pointer;
  padding-top: 3px;
  padding-bottom: 0;
  margin-top: 0;
}

.c1:hover,
.c1:focus {
  color: #006cff;
}

.c1 span {
  position: relative;
  top: 1px;
  left: -2px;
}

.c0 {
  width: 20px;
  height: 100%;
  padding-top: 0;
  margin: 0;
}

<Sorting
  activeSorter="test-string"
  category="test-string"
  sort={[Function]}
  sorting="def"
  tab={0}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.button
        activeSorter="test-string"
        category="test-string"
        onClick={[Function]}
        sorting="def"
        tabIndex={0}
        title="Current sorting: default"
      >
        <button
          className="c1"
          onClick={[Function]}
          tabIndex={0}
          title="Current sorting: default"
        >
          A/Z
        </button>
      </styled.button>
    </div>
  </styled.div>
</Sorting>
`;
